content/tracing/setup/javascript.md

```diff
@@ -0,0 +1,164 @@
+
+Next, install the Datadog Tracing library using npm:
+
+```sh
+npm install --save dd-trace
+```
+
+Finally, import the tracer and instrument your code!
+
+### Example
+
+```js
+const tracer = require('dd-trace').init()
+
+tracer
+  .trace('web.request', {
+    service: 'my_service'
+  })
+  .then(span => {
+    span.setTag('my_tag', 'my_value')
+    span.finish()
+  })
+```
+
+### Configuration Options
+
+Options can be configured as a parameter to the `init()` method or as environment variables.
+
+| Config        | Environment Variable         | Default   | Description |
+| ------------- | ---------------------------- | --------- | ----------- |
+| debug         | DD_TRACE_DEBUG               | false     | Enable debug logging in the tracer. |
+| service       | DD_SERVICE_NAME              |           | The service name to be used for this program. |
+| hostname      | DD_TRACE_AGENT_HOSTNAME      | localhost | The address of the trace agent that the tracer will submit to. |
+| port          | DD_TRACE_AGENT_PORT          | 8126      | The port of the trace agent that the tracer will submit to. |
+| flushInterval |                              | 2000      | Interval in milliseconds at which the tracer will submit traces to the agent. |
+| experimental  |                              | {}        | Experimental features can be enabled all at once using boolean `true` or individually using key/value pairs. Available experimental features: `asyncHooks`. |
+| plugins       |                              | true      | Whether or not to enable automatic instrumentation of external libraries using the built-in plugins. |
+
+## OpenTracing
+
+This library is OpenTracing compliant. Once the tracer is initialized
+it can be used as a global tracer.
+
+```js
+const tracer = require('dd-trace').init()
+const opentracing = require('opentracing')
+
+opentracing.initGlobalTracer(tracer)
+```
+
+The tracer will then be available with `opentracing.globalTracer()`.
+
+See the OpenTracing JavaScript [documentation](https://github.com/opentracing/opentracing-javascript)
+and [API](https://doc.esdoc.org/github.com/opentracing/opentracing-javascript/) for more details.
+
+**NOTE: The tracer returned by `opentracing.globalTracer()` only
+contains OpenTracing specific methods.**
+
+**NOTE: When using OpenTracing, context propagation is not handled
+automatically.**
+
+## Automatic Instrumentation
+
+APM provides out-of-the-box support for many popular integrations
+by using a plugin system. By default all built-in plugins are
+enabled. This behavior can be changed by setting the `plugins` option to `false` in the [configuration options](#configuration-options).
+
+Built-in plugins can be enabled by name:
+
+```js
+const tracer = require('dd-trace').init()
+
+tracer.use('express')
+```
+
+It is also possible to configure plugins using the same method:
+
+```js
+const tracer = require('dd-trace').init()
```

Comment:
We can merge these two examples. I would say:

Built-in plugins can be enabled by name and configured individually:
```javascript
const tracer = require('dd-trace').init()  // plugins: false?

// enable express integration
tracer.use('express')

// enable and configure postgresql integration
tracer.use('pg', {
  service: 'pg-cluster'
})
```

Note: let's use interesting service names. Instead of `my_something` or `foo`, `bar`, let's use legit names in a distributed system.